File: NetFramework/FlexGrid/VB/ColumnContextMenu/Forms/FilterEditor.vb
Imports System.Windows.Forms
Imports C1.Win.C1FlexGrid
Imports C1.DataFilter
Imports C1.Win.Ribbon

Public Class FilterEditor

    Private _defaultExpression As Expression

    Public Sub New(grid As C1FlexGrid)
        InitializeComponent()
        C1FilterEditor1.DataSource = grid.DataSource
    End Sub

    Public Async Sub SetExpression(expression As Expression)
        C1FilterEditor1.SetExpression(expression)
        Await C1FilterEditor1.ApplyFilterAsync()
    End Sub

    Public ReadOnly Property IsEmpty As Boolean
        Get
            Dim expression As CombinationExpression = C1FilterEditor1.GetExpression()
            Return expression Is Nothing Or expression.Expressions.Count = 0
        End Get
    End Property

    Private Sub FilterEditor_Shown(sender As Object, e As EventArgs) Handles MyBase.Shown
        _defaultExpression = C1FilterEditor1.GetExpression()
    End Sub

    'Clear button clicked
    Private Sub btnClear_Click(sender As Object, e As EventArgs) Handles btnClear.Click
        'Clear filter
        C1FilterEditor1.ClearFilter()
    End Sub

    'Ok button clicked
    Private Async Sub btnOk_Click(sender As Object, e As EventArgs) Handles btnOk.Click
        'Apply filter and close form
        Await C1FilterEditor1.ApplyFilterAsync()
        DialogResult = DialogResult.OK
        Close()
    End Sub

    'Cancel button clicked
    Private Sub btnCancel_Click(sender As Object, e As EventArgs) Handles btnCancel.Click
        'Undo changes and close form
        SetExpression(_defaultExpression)
        DialogResult = DialogResult.Cancel
        Close()
    End Sub
End Class

File: NetFramework/FlexGrid/CS/ColumnContextMenu/Forms/FilterEditor.cs
using System.Windows.Forms;
using C1.Win.C1FlexGrid;
using C1.DataFilter;
using C1.Win.Ribbon;

namespace ColumnContextMenu.Forms
{
    public partial class FilterEditor : C1RibbonForm
    {
        private Expression _defaultExpression;

        public FilterEditor(C1FlexGrid grid)
        {
            InitializeComponent();
            c1FilterEditor1.DataSource = grid.DataSource;
        }

        public async void SetExpression(Expression expression)
        {
            c1FilterEditor1.SetExpression(expression);
            await c1FilterEditor1.ApplyFilterAsync();
        }

        public bool IsEmpty
        {
            get
            {
                var expression = c1FilterEditor1.GetExpression() as CombinationExpression;
                return expression == null || expression.Expressions.Count == 0;
            }
        }

        private void FilterEditor_Shown(object sender, System.EventArgs e)
        {
            _defaultExpression = c1FilterEditor1.GetExpression();
        }

        // Clear button clicked
        private void btnClear_Click(object sender, System.EventArgs e)
        {
            // Clear filter
            c1FilterEditor1.ClearFilter();
        }

        // Ok button clicked
        private async void btnOk_Click(object sender, System.EventArgs e)
        {
            // Apply filter and close form
            await c1FilterEditor1.ApplyFilterAsync();
            DialogResult = DialogResult.OK;
            Close();
        }

        // Cancel button clicked
        private void btnCancel_Click(object sender, System.EventArgs e)
        {
            // Undo changes and close form
            SetExpression(_defaultExpression);
            DialogResult = DialogResult.Cancel;
            Close();
        }
    }
}
